core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/BinPackingNodeAllocatorService.java

```diff
@@ -88,7 +90,7 @@ public class BinPackingNodeAllocatorService
     private final InternalNodeManager nodeManager;
     private final Supplier<Map<String, Optional<MemoryInfo>>> workerMemoryInfoSupplier;
 
-    private final ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(2, daemonThreadsNamed("bin-packing-node-allocator"));
+    private final ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(3, daemonThreadsNamed("bin-packing-node-allocator"));
```

Comment:
why?

Reply:
we start up three routines... maybe it should be cachedThreadPool ...